### ISSUE TITLE
OutOfMemoryError: Java heap space

### ISSUE THREAD

**Issue by copperfield42** [magefree/mage#6979]
![inferno titan 1](https://user-images.githubusercontent.com/4504719/90443790-d02e9f80-e0aa-11ea-9b6d-efb501f799fc.PNG)

I try to play an _Inferno Titan_ and this happens

```
INFO  2020-08-16 19:07:40,790 MATCH started [Game] 827ba70a-c871-431f-8925-c4fb68d3dc1e(copperfield)                     =>[CALL main-74] TableController.lambda$startMatch$7 
INFO  2020-08-16 19:07:40,932 GAME started 4b23dab3-8aae-4ef0-9011-ff03ec60ec5e [Game] copperfield - Computer 3 - Computer 2 - Computer 4 =>[CALL main-74] TableController.startGame 
INFO  2020-08-16 19:21:01,591 USER REMOVE - copperfield (Disconnected)  userId: 35f6738d-fc60-41f8-9d48-069aa4256a04 [Match: 1 ] =>[CALL main-81] UserManager.lambda$null$1 
WARN  2020-08-16 19:23:59,422 Removing unhealthy tableId c16c1582-aaac-4f3e-8946-2b376fe2a33d                            =>[pool-11-thread-1] TableManager.lambda$checkTableHealthState$2 
Exception in thread "CALL main-81" java.lang.OutOfMemoryError: Java heap space
```

I suppose that the AI using the titan #6330 isn't the only source of problem with that card.

For what I understand, the AI play all possibilities to choose what it does, maybe that need some limitation for cards that are way too wide on choices like this Titan, like for example only do the full damage to only one target and not spread it around trying to find the optimal play and/or predict the play from the human player and only react to it.

-----

Additionally, some time ago (like 2 or 3 beta versions ago) I got the same problem playing an _Ezuri's Predation_ but that was only once thus far, I play it couple more times since trying to see if it happens again, but luckily no problem. 

Hopefully is the same with this.

**Comment by Alex-Vasile** [comment]
Marking this a duplicate of #9539 and tracking this class of bugs/problems there
